src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/CleanUpSensitiveLogsServiceTest.java

```diff
@@ -0,0 +1,141 @@
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+
+@ExtendWith(MockitoExtension.class)
+class CleanUpSensitiveLogsServiceTest {
+
+    private static final String KEY = "clean_up_start_date";
+
+    @Mock
+    private CFTSensitiveTaskEventLogsDatabaseService cftSensitiveTaskEventLogsDatabaseService;
+
+    @Mock
+    private SensitiveTaskEventLogsRepository sensitiveTaskEventLogsRepository;
+
+    @InjectMocks
+    private CleanUpSensitiveLogsService cleanUpSensitiveLogsService;
+
+    @Test
+    void should_return_empty_response_when_invalid_operation_name_provided() {
+        OffsetDateTime timestamp = OffsetDateTime.now();
+
+        List<TaskFilter<?>> taskFilters = createTaskFilters(KEY, timestamp);
+        TaskOperationRequest request = new TaskOperationRequest(
+            TaskOperation.builder()
+                .type(TaskOperationType.EXECUTE_RECONFIGURE)
+                .runId("123")
+                .build(),
+            taskFilters
+        );
+
+        TaskOperationResponse taskOperationResponse = cleanUpSensitiveLogsService
+            .performOperation(request);
+
+        assertNotNull(taskOperationResponse);
+
+        assertNull(taskOperationResponse.getResponseMap());
+
+    }
+
+    @Test
+    void should_clean_up_sensitive_logs() {
+        OffsetDateTime timestamp = OffsetDateTime.now();
+
+        List<TaskFilter<?>> taskFilters = createTaskFilters(KEY, timestamp);
+        TaskOperationRequest request = new TaskOperationRequest(
+            TaskOperation.builder()
+                .type(TaskOperationType.CLEANUP_SENSITIVE_LOG_ENTRIES)
+                .runId("")
+                .build(),
+            taskFilters
+        );
+
+        TaskOperationResponse taskOperationResponse = cleanUpSensitiveLogsService
+            .performOperation(request);
+
+        assertNotNull(taskOperationResponse);
+
+        int deletedRows = (int) taskOperationResponse.getResponseMap()
+            .get("deletedRows");
+
+        assertEquals(0, deletedRows);
+
+    }
+
+    @Test
+    void should_throw_exception_when_cannot_connect_to_db_during_clean_up_sensitive_logs() {
+        OffsetDateTime timestamp = OffsetDateTime.now();
+
+        List<TaskFilter<?>> taskFilters = createTaskFilters(KEY, timestamp);
+        TaskOperationRequest request = new TaskOperationRequest(
+            TaskOperation.builder()
+                .type(TaskOperationType.CLEANUP_SENSITIVE_LOG_ENTRIES)
+                .runId("")
+                .build(),
+            taskFilters
+        );
+
+        doThrow(new IllegalArgumentException("cleanup exception"))
```

Comment:
Is this the right exception to throw when unable to connect to db?

Reply:
Looks like it has been changed as part of this PR to OK to comply with the API docs: https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1552143414#WATaskManagement:POST/task/operation(internal)-SuccessfulResponses

Reply:
Created this ticket to cover this comment: https://tools.hmcts.net/jira/browse/RWA-2638